File: paseto-base/src/main/java/net/aholbrook/paseto/verification/rules/IssuedBy.java
package net.aholbrook.paseto.verification.rules;

import net.aholbrook.paseto.Token;
import net.aholbrook.paseto.exception.verification.rules.IncorrectIssuerException;
import net.aholbrook.paseto.util.StringUtils;
import net.aholbrook.paseto.verification.PasetoVerificationContext;

public class IssuedBy implements Rule {
	public final static String NAME = "ISSUED_BY";

	private final String issuer;

	/**
	 * Verifies that the token Issuer (iss) claim matches the given value.
	 *
	 * @param issuer The expected issuer of the token.
	 */
	public IssuedBy(String issuer) {
		this.issuer = StringUtils.ntes(issuer);
	}

	@Override
	public String name() {
		return NAME;
	}

	@Override
	public void check(Token token, PasetoVerificationContext context) {
		String tokenIssuer = StringUtils.ntes(token.getIssuer());

		if (!issuer.equals(tokenIssuer)) {
			throw new IncorrectIssuerException(issuer, token.getIssuer(), NAME, token);
		}
	}
}

File: paseto/src/main/kotlin/net/aholbrook/paseto/rules/IssuedBy.kt
package net.aholbrook.paseto.rules

import net.aholbrook.paseto.Token
import net.aholbrook.paseto.crypto.constantTimeEquals
import net.aholbrook.paseto.exception.IncorrectIssuerException
import net.aholbrook.paseto.exception.MissingClaimException

/**
 * Verifies that the token Issuer (iss) claim matches the given value.
 *
 * @param issuer The expected issuer of the token.
 */
data class IssuedBy(val issuer: String) : Rule {
    override operator fun invoke(token: Token, mode: Rule.Mode, currentResults: Map<Rule, RuleResult>) {
        if (token.issuer.isNullOrEmpty()) {
            throw MissingClaimException("iss", token)
        }

        if (!token.issuer.constantTimeEquals(issuer)) {
            throw IncorrectIssuerException(issuer, token.issuer, this, token)
        }
    }
}
